pkg/controller/v1beta1/inferenceservice/reconcilers/deployment/deployment_reconciler.go

```diff
@@ -98,6 +98,11 @@ func (r *DeploymentReconciler) checkDeploymentExist(client client.Client) (const
 	//existed, check equivalence
 	//for HPA scaling, we should ignore Replicas of Deployment
 	ignoreFields := cmpopts.IgnoreFields(appsv1.DeploymentSpec{}, "Replicas")
+	// Do a dry-run update. This will populate our local deployment object with any default values
+	// that are present on the remote version.
+	if err := client.Update(context.TODO(), r.Deployment, kclient.DryRunAll); err != nil {
+		return constants.CheckResultUnknown, nil, err
```

Comment:
If there are errors, that means dry-run failed right?

Reply:
let's add a log line here for better debugging experience in case dry run fails.